platform-channels.md

```diff
@@ -171,17 +170,19 @@ refreshing the valuer.
 ```dart
 @override
 Widget build(BuildContext context) {
-  return new Material(
-    child: new Center(
-      child: new Column(
-        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-        children: [
-          new RaisedButton(
-            child: new Text('Get Battery Level'),
-            onPressed: _getBatteryLevel,
-          ),
-          new Text(_batteryLevel),
-        ],
+  return new MaterialApp(
+    home: new Material(
```

Comment:
We probably don't need the `Material` then.

Reply:
If you mean something like this:

```dart
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
        home: new Center(
          child: new Column(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: [
              new RaisedButton(
                child: new Text('Get Battery Level'),
                onPressed: _getBatteryLevel,
              ),
              new Text(_batteryLevel),
            ],
          ),
      ),
    );
  }
```

I actually tried it however it looks like `MaterialButton` widgets require a `Material` widget ancestor.

```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following assertion was thrown building MaterialButton(dirty, state:
No Material widget found.
MaterialButton widgets require a Material widget ancestor.
In material design, most widgets are conceptually "printed" on a sheet of material. In Flutter's
material library, that material is represented by the Material widget. It is the Material widget
that renders ink splashes, for instance. Because of this, many material library widgets require that
there be a Material widget in the tree above them.
To introduce a Material widget, you can either directly include one, or use a widget that contains
Material itself, such as a Card, Dialog, Drawer, or Scaffold.
The specific widget that could not find a Material ancestor was:
MaterialButton
The ownership chain for the affected widget is:
MaterialButton ← RaisedButton ← Column ← Center ← _ModalScopeStatus ←
PageStorage-[GlobalKey#f01fd] ← RepaintBoundary ← Stack ← _CupertinoBackGestureDetector ←
DecoratedBox ← ⋯
```

Reply:
Ah, right. `MaterialApp` is usually used with a `Scaffold` which is what supplies the `Material` I guess. Nevermind then.

Reply:
For the record, the corresponding code in https://github.com/flutter/flutter/blob/master/examples/platform_channel/lib/main.dart does wrap the `Material` widget in a `MaterialApp`. It just does so in `main` rather than in the `build` method from which the code for the web page was copied.